app/controllers/exam_templates_controller.rb

```diff
@@ -9,6 +9,52 @@ def index
     @exam_templates = @assignment.exam_templates
   end
 
+  # Creates a new instance of the exam template.
+  def create
+    assignment = Assignment.find(params[:assignment_id])
+    # retrieving exam template file
+    new_uploaded_io = params[:create_template][:file_io]
+    # getting filename
+    filename = params[:create_template][:filename]
```

Comment:
Do not allow the user to input a filename on the form. The filename used should be the one of the uploaded file (from "file_io").